tools/clang/unittests/HLSL/ValidationTest.cpp

```diff
@@ -637,6 +635,7 @@ TEST_F(ValidationTest, WhenDepthNotFloatThenFail) {
 }
 
 TEST_F(ValidationTest, BarrierFail) {
+  if (!m_CompilerPreservesBBNames) return;    // This test requires name preservation to succeed currently.
```

Comment:
I think we definitely want to preserve the behavior that non-zero exit code means failure. If any non-pass result is treated as failure then skipping the test like this seems to be the best bet.